lib/collection/src/collection/resharding_update_pre_filter.rs

```diff
@@ -62,32 +65,33 @@ impl ReshardingUpdatePreFilter {
             ReshardingDirection::Down => !is_target_shard,
         };
 
-        // Shard that will be *sending* migrated points during resharding:
-        // - *non* target shards during resharding *up*
-        // - *target* shard during resharding *down*
-        let is_sender_shard = !is_receiver_shard;
-
-        // We apply resharding pre-filter:
-        //
-        // - on *receiver* shards during `MigratingPoints` stage
-        // - and on *sender* shards during `ReadHashRingCommitted` stage when resharding *up*
-        let should_filter = (is_receiver_shard && state.stage == ReshardStage::MigratingPoints)
-            || (is_sender_shard
-                && state.stage >= ReshardStage::ReadHashRingCommitted
-                && state.direction == ReshardingDirection::Up);
-
-        if !should_filter {
+        // After write hash ring commit, we stop filtering
+        if state.stage >= ReshardStage::WriteHashRingCommitted {
```

Comment:
I never explained this explicitly, but I don't think we can do this. E.g.:

- Node A already received `WriteHashRingCommitted` message from consensus
  - and so it does not apply pre-filter locally anymore
- but Node B did not yet received `WriteHashRingCommitted` from consensus
  - and so it will still send all operations to new and *old* shards both
- now, user sends `upsert point 1337` to Node B
  - Node B forwards this update to *old* shard on Node A
  - and Node A inserts this point into *old* shard, because it does not apply pre-filter anymore

Basically, I think we should apply pre-filter until resharding is finished.

Reply:
> Basically, I think we should apply pre-filter until resharding is finished.

Fine by me. We can eventually fine tune it if desired.